### ISSUE TITLE
Field type is not validated if a record or bean class is not set

### ISSUE THREAD

**Issue by GoogleCodeExporter** [unlockha/beanio#18]
```
A mapping file such as the following, should still validate the 'fileDate' 
field in the 'header' record is a valid date in MMddyyyy format (even though 
the header record is not returned from BeanWriter because the 'class' attribute 
is not set).

  <stream name="stream4" format="csv">
    <record name="header" minOccurs="1" maxOccurs="1">
      <field name="fileDate" type="date" format="MMddyyyy" />
      <field name="fileName" />
    </record>
    <record name="record" class="map">
      <field name="field1" />
      <field name="field2" />
    </record>
  </stream>

This has likely been a defect in all previous versions.

```

Original issue reported on code.google.com by `kevin.s...@gmail.com` on 16 Sep 2011 at 9:40

**Comment by GoogleCodeExporter** [comment]
```
Similar behavior is observed if ignore="true"
```

Original comment by `kevin.s...@gmail.com` on 17 Sep 2011 at 3:47

**Comment by GoogleCodeExporter** [comment]
```
Fixed for 1.2.1.
```

Original comment by `kevin.s...@gmail.com` on 17 Sep 2011 at 2:36
- Added labels: **Milestone-1.2.1**

**Comment by GoogleCodeExporter** [comment]
Original comment by `kevin.s...@gmail.com` on 21 Sep 2011 at 10:53
- Changed state: **Fixed**
